t/issue/1508.t

```diff
@@ -0,0 +1,165 @@
+
+=head1 NAME 
+
+issue/1508.t  - Check that the `needs()` function correctly propogates run-time task arguments
+
+=head1 DESCRIPTION
+
+Check that the `needs()` function does indeed correctly propogate 
+run-time task arguments (%params and @args) from the calling task down to the "needed" tasks.
+
+=head1 DETAILS
+
+  * AUTHOR / DATE : [tabulon]@[2021-09-26]
+  * RELATES-TO    : [github issue #1508](https://github.com/RexOps/Rex/issues/1508#issue-1007457392)
+  * INSPIRED from : t/needs.t
+
+=cut
+
+use Test::More;
+use Rex::Commands;
+
+{
+
+  package T; # Helper package (for cutting down boilerplate in tests)
+  use Storable;
+
+  sub track_taskrun {
+    my %opts = ref $_[-1] eq 'HASH'
+      ? %{
+      ;
+      pop
+      }
+      : ();
```

Comment:
Based on the calls in this file, it seems that the first argument is always a scalar value, and the last argument is always a hash reference. Does it make sense to still keep this unpack logic around?

Maybe this works too:

```perl
my ($called_task, %opts) = @_;
```

That way we don't need the for loop either. What do you think?

(Same goes for `check_needed` below)

Reply:
Perhaps... 

The intention was the ability to do the following, when you have more than one needed task : 

```perl
needs ("task1", "task2", "task3")
check_needed("task1", "task2", "task3", \%opts) 
````
Currently, it looks like we don't have that case (with several needed tasks) in the tests. Maybe we should ?

Reply:
For `track_taskrun`, it's an easier call; yes we could get rid of some of the logic, as you suggest.

Reply:
> Currently, it looks like we don't have that case (with several needed tasks) in the tests. Maybe we should ?

I think that's what confused me initially. We don't have the multiple-tasks case tested now, so it might be premature optimizing.

But your're right, in the long run we probably should cover that use case too :+1: